_assets/scripts/commit_check.sh

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+
+set -e
+set -o pipefail
+
+parse_commits() {
+    start_commit=${1:-origin/develop}
+    end_commit=${2:-HEAD}
+    is_breaking_change=false
+
+    echo "checking commits between: $start_commit $end_commit" >&2
+    # Run the loop in the current shell using process substitution
+    while IFS= read -r message || [ -n "$message" ]; do
+        # Check if commit message follows conventional commits format
+        if [[ $message =~ ^(build|chore|ci|docs|feat|fix|perf|refactor|revert|style|test)(\(.*\))?(\_|!):.*$ ]]; then
+            # Check for breaking changes
+            if [[ ${BASH_REMATCH[3]} == *'!'* ]]; then
+                is_breaking_change=true
+                break
+            fi
+        else
+            echo "Commit message \"$message\" is not well-formed. Aborting merge."
+            # Uncomment the line below if you want to exit on an invalid commit message
```

Comment:
maybe we could add a link to conventional commits here to educate the user on why this failed